### PR TITLE
fix-issue-197: Make disabled fields untabable

### PR DESCRIPTION
**Pull request by zzacharo**
Signed-off-by: Zacharias Zacharodimos <zacharias.zacharodimos@cern.ch>

### CODE REVIEW

File: src/primitive-field/primitive-field.component.ts [hunk 1]
```diff
@@ -105,7 +105,11 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
   }
 
   get tabIndex(): number {
-    return this.tabIndexService.getElemTabIndex(this.pathString);
+    if (this.schema['disabled'] && this.schema['disabled'] === true) {
```
Comment (jmartinm) [inline src/primitive-field/primitive-field.component.ts]: should be enough with:

```javascript
if (this.schema['disabled'] === true)
```

**Comment by jmartinm** [comment]
Can you check the `Admin mode`? It does not enable the Disabled fields and I think it was doing it before, @harunurhan

**Comment by zzacharo** [comment]
yes i noticed this...in master branch is working correctly?

**Comment by jmartinm** [comment]
> yes i noticed this...in master branch is working correctly?

Please test, and if this PR broke it add a test to make sure it does not break in the future.

**Comment by zzacharo** [comment]
No it doesn't work in master too.

File: src/shared/services/schema-fixer.service.ts [hunk 2]
```diff
@@ -172,6 +192,11 @@ export class SchemaFixerService {
           };
         }
       });
+    // copy disabled attribute inside fixedSchema because it
+    // is outside anyOf element and is ignored
+    if (schema['disabled']) {
+      Object.assign(fixedSchema, {disabled: schema['disabled']});
```
Comment (harunurhan) [inline src/shared/services/schema-fixer.service.ts]: `fixedSchema['disabled'] = true` would simply work.

File: src/primitive-field/primitive-field.component.ts [hunk 3]
```diff
@@ -105,7 +105,7 @@ export class PrimitiveFieldComponent extends AbstractFieldComponent implements O
   }
 
   get tabIndex(): number {
-    return this.tabIndexService.getElemTabIndex(this.pathString);
+    return this.schema['disabled'] === true ? -1 : this.tabIndexService.getElemTabIndex(this.pathString);
```
Comment (harunurhan) [inline src/primitive-field/primitive-field.component.ts]: `this.schema['disabled'] ? -1 : ...`

Comment (zzacharo) [inline src/primitive-field/primitive-field.component.ts]: In theory this would also return -1 if this.schema['disabled'] is just defined.

Comment (harunurhan) [inline src/primitive-field/primitive-field.component.ts]: `disabled` could be either `true`or `false`, so for my comment, when it's true, it'll return `-1` and when it's `false` or `undefined` it'll return real `tabindex`.

**Pull request merged**